test_runner/regress/test_download_extensions.py

```diff
@@ -245,3 +249,57 @@ def test_remote_extensions(
                 log.info(f"Deleted {file}")
             except FileNotFoundError:
                 log.info(f"{file} does not exist, so cannot be deleted")
+
+
+"""
+This tests against the actual infra for real S3
+Note in particular that we don't need to set up a bucket (real or mock)
+because we are testing the files already uploaded as part of CI/CD
+"""
+
+
+def test_remote_extensions_in_bucket(neon_env_builder: NeonEnvBuilder):
+    neon_env_builder.enable_remote_storage(
+        remote_storage_kind=RemoteStorageKind.REAL_S3,
+        test_name="test_remote_extensions_in_bucket",
+        enable_remote_extensions=False,  # we don't enable remote extensions here; instead we use the real bucket
+    )
+    neon_env_builder.num_safekeepers = 3
+    env = neon_env_builder.init_start()
+    tenant_id, _ = env.neon_cli.create_tenant()
+    env.neon_cli.create_timeline("test_remote_extensions_in_bucket", tenant_id=tenant_id)
+
+    # Start a compute node and check that it can download the extensions
+    # and use them to CREATE EXTENSION and LOAD 'library.so'
+    remote_ext_config = {
+        "bucket": "neon-dev-extensions-eu-central-1",
+        "region": "eu-central-1",
+        "endpoint": None,
+        "prefix": "5412197734",  # build tag
+    }
+    endpoint = env.endpoints.create_start(
+        "test_remote_extensions_in_bucket",
+        tenant_id=tenant_id,
+        remote_ext_config=json.dumps(remote_ext_config),
+    )
+
+    with closing(endpoint.connect()) as conn:
+        with conn.cursor() as cur:
+            cur.execute("SELECT * FROM pg_available_extensions")
+            all_extensions = [x[0] for x in cur.fetchall()]
+            # check that one of the extensions was downloaded
+            assert "pg_surgery" in all_extensions
+
+            log.info(f"DOWNLOADED EXTENSIONS: {all_extensions}")
+
+            cur.execute("CREATE EXTENSION pg_surgery")
+            cur.execute("SELECT extname FROM pg_surgery")
```

Comment:
should it be this?

```suggestion
            cur.execute("SELECT extname FROM pg_extension")
```

Reply:
No, I want to  download the `pg_surgery` extension https://www.postgresql.org/docs/current/pgsurgery.html

Reply:
then you need to run `CREATE EXTENSION pg_surgery`

Reply:
oh!